extensions/vault/runtime/RestClientVaultClient.java

```diff
@@ -0,0 +1,126 @@
+package io.quarkus.vault;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation;
+
+import io.quarkus.vault.proxy.dto.VaultDbCreds;
+import io.quarkus.vault.proxy.dto.VaultHealth;
+import io.quarkus.vault.proxy.dto.VaultInit;
+import io.quarkus.vault.proxy.dto.VaultInitBody;
+import io.quarkus.vault.proxy.dto.VaultKubernetesAuth;
+import io.quarkus.vault.proxy.dto.VaultKubernetesAuthBody;
+import io.quarkus.vault.proxy.dto.VaultKvSecretV1;
+import io.quarkus.vault.proxy.dto.VaultKvSecretV2;
+import io.quarkus.vault.proxy.dto.VaultLeasesBody;
+import io.quarkus.vault.proxy.dto.VaultLeasesLookup;
+import io.quarkus.vault.proxy.dto.VaultLookupSelf;
+import io.quarkus.vault.proxy.dto.VaultRenewLease;
+import io.quarkus.vault.proxy.dto.VaultRenewSelf;
+import io.quarkus.vault.proxy.dto.VaultRenewSelfBody;
+import io.quarkus.vault.proxy.dto.VaultSealStatus;
+import io.quarkus.vault.proxy.dto.VaultUserPassAuth;
+import io.quarkus.vault.proxy.dto.VaultUserPassAuthBody;
+
+public class RestClientVaultClient implements VaultClient {
```

Comment:
@vsevel you are right, lets drop till we get back to the idea of using this type of Rest client

Reply:
and that solves this issue by the way: https://github.com/quarkusio/quarkus/pull/2897#issuecomment-510992873